src/Microsoft.OData.Core/UriParser/Visitors/PathSegmentToContextUrlPathTranslator.cs

```diff
@@ -180,6 +180,36 @@ public override string Translate(CountSegment segment)
             return string.Empty;
         }
 
+        /// <summary>
+        /// Translate a FilterSegment
+        /// </summary>
+        /// <param name="segment">the segment to Translate</param>
+        /// <returns>Defined by the implementer.</returns>
+        public override string Translate(FilterSegment segment)
+        {
+            return "/" + segment.FullSegment;

```

Comment:
>return "/" + segment.FullSegment; [](start = 12, length = 33)

Filter segment should not appear in contextUrl; just return string.Empty.

Reply:
Gotcha.